books/views.py

```diff
@@ -73,15 +71,10 @@ def get(self, request, pk):
     def post(self, request, pk):
         book = Book.objects.get(pk=pk)
         form = self.form_class(request.POST)
+        form.instance = book
```

Comment:
Tip: You can pass the instance when you initialise the form:
```
form = self.form_class(request.POST, instance=book)
```

When your form contains fields and you use it to "update' something, then it's preferable to do it this way, because Django will use it in the __init__ to create the initial (pre-filled) data. If you pass it later it may be too late 🔥

Reply:
Ah good to know!